### PR TITLE
Removing deprecations in preparation of v2.0

### PR DESCRIPTION
**Pull request by bsipocz**
This is a rather controversial PR, and I'm very much interested in what you all think.

I've gone ahead and removed (hopefully) every deprecations that was present in v1.3. According to [APE2](https://github.com/astropy/astropy-APEs/blob/master/APE2.rst), we ought to keep deprecations around for one feature release. Since 2.0 will be our new LTS, I suggest to remove all of these deprecation now rather than dragging them along for 2 more years.

Obviously submodule maintainers have the insight whether we need to keep any of these for a longer cool off period.

(This will need a changelog that I add once we agreed on the deprecations that are needed to be kept further in core.)

### CODE REVIEW

**Comment by nden** [comment]
@bsipocz Removing the deprecations in modeling and wcs is fine. Thanks for taking care of this.

**Comment by mwcraig** [comment]
@bsipocz -- the nddata changes look good, thanks for doing this!

File: astropy/io/fits/hdu/hdulist.py [hunk 1]
```diff
@@ -1068,31 +1061,6 @@ def _try_while_unread_hdus(self, func, index, *args, **kwargs):
...
-                    even though hdulist[2] had not been read yet.  Instead
-                    do:
-
-                        >>> print(hdulist[2].header)
-                        >>> hdulist.close()
-
-                    or open the file with lazy_load_hdus=False to read all
-                    the HDUs into memory immediately upon opening the file.
-                    """), AstropyDeprecationWarning)
-
```
Comment (saimn) [inline astropy/io/fits/hdu/hdulist.py]: This one will be removed as part of #6082.
Otherwise looks good :+1:

**Comment by mhvk** [comment]
@bsipocz - removals in `angles` and `units` are both fine; thanks!

**Comment by MSeifert04** [comment]
nddata and io.fits look okay .

File: astropy/io/fits/hdu/hdulist.py (hunk 1 above)
Comment (MSeifert04) [inline astropy/io/fits/hdu/hdulist.py]: My feeling is that this should only be removed **if and only if** #6082 is merged. So I would suggest not removing it in here.

**Comment by bsipocz** [comment]
Thanks everyone for the feedbacks. There are still a few submodules that needs to be checked:

@aconley, @nhmc  - Could either of you confirm please that the cosmology deprecations can be removed?

@pllim - Could you check  the ``io.votable`` part?

@larrybradley @astrofrog - Could you have a look at the visualization removal?

**Comment by eteq** [comment]
Hmm, I worry a little that `rotation_matrix` is used in the wild. @adrn, what do you think?

(Other than that I'm happy with this)

**Comment by adrn** [comment]
Well, I was a user, but I updated my code when I started getting deprecation warnings, so the system worked :) 

I think it's probably fine to remove by now...

**Comment by pllim** [comment]
@bsipocz , `io.votable` LGTM. I can't imagine people would still use the old `voexceptions`.

**Comment by mhvk** [comment]
@eteq - just in case you were confused by the two sets of deprecation messages: `rotation_matrix` was moved to `utils`, not removed altogether, and here we're just removing it from the old location.

**Comment by larrybradley** [comment]
@bsipocz `visualization` LGTM.  Thanks!

**Comment by nhmc** [comment]
I will check the cosmology cases tonight, unless @aconley beats me to it.

**Comment by aconley** [comment]
I'm on vacation ATM, so I think you're gonna get there first.

**Comment by nhmc** [comment]
@bsipocz - Removing the cosmology deprecations seems fine to me.

**Comment by bsipocz** [comment]
Now that this has got the green light from every affected subpackage, I'm going ahead and merge. Thanks for the review!

**Pull request merged**